server/sonar-server/src/test/java/org/sonar/server/permission/ws/AddUserActionTest.java

```diff
@@ -160,20 +129,22 @@ public void fail_when_project_uuid_is_unknown() throws Exception {
   public void fail_when_project_permission_without_project() throws Exception {
     expectedException.expect(BadRequestException.class);
 
-    ws.newPostRequest(CONTROLLER, ACTION)
-      .setParam(PARAM_USER_LOGIN, "ray.bradbury")
+    loginAsAdmin();
+    wsTester.newPostRequest(CONTROLLER, ACTION)
+      .setParam(PARAM_USER_LOGIN, user.getLogin())
       .setParam(PARAM_PERMISSION, UserRole.ISSUE_ADMIN)
       .execute();
   }
 
   @Test
   public void fail_when_component_is_not_a_project() throws Exception {
+    componentTester.insertComponent(newFileDto(newProjectDto("project-uuid"), null, "file-uuid"));
+
     expectedException.expect(BadRequestException.class);
-    insertComponent(newFileDto(newProjectDto("project-uuid"), null, "file-uuid"));
-    commit();
 
-    ws.newPostRequest(CONTROLLER, ACTION)
-      .setParam(PARAM_USER_LOGIN, "ray.bradbury")
+    loginAsAdmin();
```

Comment:
`expectedException.expect` should right before the statement supposed to raise the exception
as such, `loginAsAdmin()` should be before `expectedException.expect`, not after it